provider/vsphere/internal/vsphereclient/client.go

```diff
@@ -438,8 +475,31 @@ func (c *Client) cloneVM(
 	return object.NewVirtualMachine(c.client.Client, info.Result.(types.ManagedObjectReference)), nil
 }
 
+func (c *Client) getDiskWithFileBacking(
+	ctx context.Context,
+	vm *object.VirtualMachine,
+) (*types.VirtualDisk, types.BaseVirtualDeviceFileBackingInfo, error) {
+	var mo mo.VirtualMachine
+	if err := c.client.RetrieveOne(ctx, vm.Reference(), []string{"config.hardware"}, &mo); err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	for _, dev := range mo.Config.Hardware.Device {
+		dev, ok := dev.(*types.VirtualDisk)
+		if !ok {
+			continue
+		}
+		backing, ok := dev.Backing.(types.BaseVirtualDeviceFileBackingInfo)
+		if !ok {
+			continue
+		}
+		return dev, backing, nil
```

Comment:
so there can't be more than one?
we just return the first one?

Reply:
We create this machine from a template, I think there can only be one.